front/src/applications/operationalStudies/utils.ts

```diff
@@ -209,3 +214,79 @@ export const convertDepartureTimeIntoSec = (departureTime: string) => {
 
 export const isInvalidName = (name?: string | null) =>
   !name || name.length > SMALL_INPUT_MAX_LENGTH;
+
+/**
+ * Check if the scheduled points are honored with a trainSchedule and a trainSummary
+ * @param trainResult
+ * @param trainSummary
+ * @returns true if the scheduled points are not honored
+ */
+export const isScheduledPointsNotHonored = (
+  trainSchedule: TrainScheduleResult,
+  trainSummary: Extract<SimulationSummaryResult, { status: 'success' }>
+): boolean => {
+  if (trainSummary.path_item_times_final.length !== trainSchedule.path.length) {
+    throw new Error('Asserstion failed');
+  }
+  if (!trainSchedule.schedule) return false;
+  return trainSchedule.schedule.some((schedule, index) => {
+    if (schedule.arrival === null) return false;
+    // need to add +1 because origin will never be in schedules
+    return (
+      Math.abs(
+        sToMs(ISO8601Duration2sec(schedule.arrival as string)) -
+          trainSummary.path_item_times_final[index + 1]
+      ) >= TOLERANCES
+    );
+  });
+};
+
+export const pathItemMapping = (trainResult: TrainScheduleResult) => {
+  const pathItemMap: { [key: string]: number } = {};
+
+  trainResult.path.forEach((pathItem, index) => {
+    if (pathItemMap[pathItem.id]) throw new Error('Assertion failed');
+    pathItemMap[pathItem.id] = index;
+  });
+
+  return pathItemMap;
+};
+
+/**
+ * Check if the train is too fast with a trainSchedule and a trainSummary
+ * @param trainResult
+ * @param trainSummary
+ * @returns true if the train is too fast
+ */
+export const isToofast = (
+  trainSchedule: TrainScheduleResult,
+  trainSummary: Extract<SimulationSummaryResult, { status: 'success' }>
+): boolean => {
+  if (
+    trainSummary.path_item_times_final.length !== trainSummary.path_item_times_provisional.length
+  ) {
+    throw new Error('Assertion failed');
+  }
+
+  const toCheckPathItemIds = new Set<string>(trainSchedule.margins?.boundaries);
```

Comment:
You can remove the string here, typescript knows that it will be a set of strings ;)
```suggestion
  const toCheckPathItemIds = new Set(trainSchedule.margins?.boundaries);
```

Reply:
Done